src/main/scala/carldata/series/TimeConverter.scala

```diff
@@ -40,19 +40,50 @@ object TimeConverter {
       }
     }
 
+    def isAny(c: CronElement): Boolean = {
+      c match {
+        case AnyElement => true
+        case _ => false
+      }
+    }
+
+    def setDayOfWeek(t1: Int, ct2: CronElement, dt: LocalDateTime): LocalDateTime = {
+      ct2 match {
+        case nt2: NumberElement =>
+          val t2 = nt2.v
+          if (t1 == t2) dt
+          else if (t1 > t2) dt.minusDays(t1 - t2)
+          else dt.minusDays(7 - (t2 - t1))
+        case _ => dt
+      }
+    }
+
     val t: Seq[Int] = Seq(
       floor(dt.getMinute, c.minutes)
       , floor(dt.getHour, c.hour)
       , floor(dt.getDayOfMonth, c.dayOfMonth)
       , floor(dt.getMonthValue, c.month)
       , floor(dt.getDayOfWeek.getValue, c.dayOfWeek)
     )
-    var res = if (t(0) > 0) dt.withMinute(t(0)) else dt.minusHours(1).withMinute(t(0))
-    res = if (t(1) > 0) res.withHour(t(1)) else res.minusDays(1).withHour(Math.abs(t(1))).withMinute(59)
-    res = if (t(3) > 0) {
-      if (t(3) == dt.getMonthValue) res.withMonth(t(3))
-      else res.withMonth(Math.abs(t(3))).`with`(TemporalAdjusters.lastDayOfMonth()).withHour(23).withMinute(59)
-    } else res.minusYears(1).withMonth(Math.abs(t(3))).`with`(TemporalAdjusters.lastDayOfMonth()).withHour(23).withMinute(59)
+
+    var res = if (t(3) > 0) {
+      if (t(3) == dt.getMonthValue) dt.withMonth(t(3))
+      else dt.withMonth(Math.abs(t(3))).`with`(TemporalAdjusters.lastDayOfMonth()).withHour(23).withMinute(59)
+    } else dt.minusYears(1).withMonth(Math.abs(t(3))).`with`(TemporalAdjusters.lastDayOfMonth()).withHour(23).withMinute(59)
+
+    if (!isAny(c.dayOfMonth)) res = if (t(2) > 0) {
```

Comment:
Replace condition and add curly brackets for better readability.
```scala
if(c.dayOfMonth != AnyElement) {
...
}
```
parseList
parseAny